### ISSUE TITLE
Wrong mipmaps  (DDS with alpha)

### ISSUE THREAD

**Issue by Bxaa** [ImageMagick/ImageMagick#3701]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched [open](https://github.com/ImageMagick/ImageMagick/issues) and [closed](https://github.com/ImageMagick/ImageMagick/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported
- [X] I have verified that I am using the latest version of ImageMagick

### ImageMagick version

ImageMagick-7.0.11-12-portable-Q16-HDRI-x64

### Operating system

Windows

### Operating system, version and so on

Win10x64

### Description

_All IM version with bugged mipmap resizing. 
You should add -separate -combine implementation by default for all DDS with alpha while resizing mips (Mip images lost pixels at color channel while resizing)_

_You get this issue if use resize for DDS with alpha without -separate  -combine
-separate  -combine not affect to mipmap resizing_

### Steps to Reproduce

f

**Comment by fmw42** [comment]
Please post a proper full command demonstrating the issue.

**Comment by Bxaa** [comment]
Example:
convert test.DDS -separate -colorspace RGB -define filter:Lanczos -resize 50% -define dds:cluster-fit=true -colorspace srgb -combine test.DDS
Output DDS with artefacted mips (-separate not affect to mipmaps)
(You can see it an any game)

Broken normal map (DXT5 Normal map with alpha)
<a href="https://imgbb.com/"><img src="https://i.ibb.co/sVv9RS9/b4i4RLg.png" alt="b4i4RLg" border="0"></a>

Aslo: Wrong calculate cout of mips by new size?

You shoul check it. You should check it. Same issue was present in directx tex

**Comment by fmw42** [comment]
Please post your input DDS file so the developers can test with it to duplicate your result.

**Comment by Bxaa** [comment]
Test DDS:
https://drive.google.com/file/d/11y1Oo74WCcvx4UZVZZdUT84or958bggp/view?usp=sharing

**Comment by fmw42** [comment]
Your output above does not look anything like what I get.

**Comment by Bxaa** [comment]
It does not matter. This happens with any textures of this type.

**Comment by Bxaa** [comment]
Extract mipmaps leveled image from texture and look at it
**Are you programmer?**

**Comment by fmw42** [comment]
I do not know what to expect in the result. They look like they have transparency. I do not see anything different when doing compression=dx2 without resize. But I am not familiar with DDS files. So I will leave it to the IM developers. No I am not a programmer.

**Comment by Bxaa** [comment]
Ok

**Comment by Bxaa** [comment]
> I do not know what to expect in the result. They look like they have transparency. I do not see anything different when doing compression=dx2 without resize. 

Of course, you won't see anything like that (Look at color channel of mipmaps images)

**Comment by fmw42** [comment]
You should post an exact command that you expect people to run to produce your results and make it easy to see what is happening. I am not an expert in DDS files. So I ran the command you gave which did not match your posted example.

**Comment by fmw42** [comment]
If I turn alpha off and save to PNG, this is what I get:

Input:

![test](https://user-images.githubusercontent.com/13458218/118903199-14fe9580-b8cc-11eb-9566-ea38aad90183.png)

`magick DDS:test_normal_n.dds -separate -colorspace rgb -define filter:Lanczos -define dds:compression=dx2 -define dds:cluster-fit=true -colorspace srgb -combine DDS:test_final2.DDS`

`DDS:test_final2.DDS -alpha off test_final2.png`

![test_final2](https://user-images.githubusercontent.com/13458218/118903257-3e1f2600-b8cc-11eb-83ef-ab4abb4d59b3.png)

So this clearly shows a difference. Is this the issue?

**Comment by Bxaa** [comment]
> If I turn alpha off and save to PNG, this is what I get
This is main image, not MIPMAP
**Issue with mipmaps**
https://en.wikipedia.org/wiki/Mipmap

**Comment by Bxaa** [comment]
And the fact that they are broken is a fact. 
Only one solution is : Remove all mip array and then rebuilt it with not IM

**Comment by Bxaa** [comment]
Look at this:
https://youtu.be/VY9NGEaNT0M
Only removing and rebuilding mipmaps via standalone tools solve this issue
Bugged mips at all dxt compression with alpha (dxt 2-5)
